### ISSUE TITLE
`UR_UNINIT_READ` inconsistent between compound `|=` and the equivalent `x = x | …` in a constructor

### ISSUE THREAD

**Issue by Chordrain** [spotbugs/spotbugs#4139]
### SpotBugs version

SpotBugs 4.9.8 (JDK 21, javac 21, source/target 21)

### Check name (bug pattern)

`UR_UNINIT_READ` — Uninitialized read of field in constructor (CORRECTNESS, rank 3)

### Description

A constructor reads field `m_iType` (an `int`, still at its default value `0`) before assigning it, via a compound bitwise-or assignment. The Java Language Specification defines `m_iType |= e` to be exactly equivalent to `m_iType = m_iType | (e)` — both read `m_iType`, OR it with `e`, and store the result. Yet `UR_UNINIT_READ` fires only on the explicit form, not on the compound form.

### Reproducer

**Before:**

```java
class NonShortCircuit {
    public static final int BIT0 = 1;
    protected int m_iType;

    public NonShortCircuit(boolean available) {
        m_iType |= available ? BIT0 : 0;            // BEFORE: NOT reported
    }
}
```

**After:**

```java
class NonShortCircuit {
    public static final int BIT0 = 1;
    protected int m_iType;

    public NonShortCircuit(boolean available) {
        m_iType = m_iType | (available ? BIT0 : 0);  // AFTER: reported
    }
}
```

### Expected behavior

Both forms perform the same uninitialized read of `m_iType` in the constructor, so `UR_UNINIT_READ` should report identically on both (either both fire or neither).

### Actual behavior

- BEFORE (`m_iType |= …`): **0** reports.
- AFTER  (`m_iType = m_iType | …`): **1** report — `Uninitialized read of m_iType in new NonShortCircuit(boolean)`.
